File: app/src/main/java/com/dippindab/alarmclockorganizerapp/AlarmFolder.java
package com.dippindab.alarmclockorganizerapp;

import android.provider.AlarmClock;

import java.util.ArrayList;

public class AlarmFolder {

    private ArrayList<AlarmClock> folder;
    public String name;
    public int id;

    public void setAllAlarms(boolean on) {

    }

    public void modifyAllAlarmTimes(int minutes) {

    }

    public void addAlarm(AlarmClock alarm) {
        folder.add(alarm);
    }

    public void deleteAlarm(AlarmClock alarm) {
        folder.remove(alarm);
    }

    public void setFolder(ArrayList<AlarmClock> folder) {
        this.folder = folder;
    }
}

File: app/src/main/java/com/dippindab/alarmclockorganizerapp/AlarmFolder.kt
package com.dippindab.alarmclockorganizerapp

import android.provider.AlarmClock

import java.util.ArrayList

class AlarmFolder {

    private var folder: ArrayList<AlarmClock>? = null
    var name: String? = null
    var id: Int = 0

    fun setAllAlarms(on: Boolean) {

    }

    fun modifyAllAlarmTimes(minutes: Int) {

    }

    fun addAlarm(alarm: AlarmClock) {
        folder!!.add(alarm)
    }

    fun deleteAlarm(alarm: AlarmClock) {
        folder!!.remove(alarm)
    }

    fun setFolder(folder: ArrayList<AlarmClock>) {
        this.folder = folder
    }
}
